### PR TITLE
Fix/ob-27_fix-tech-debts

### PR DESCRIPTION
**Pull request by ain-odin**
1. Changed the folder name of `src/components/PromoOrangeHeader` -> `PromoHeader` 
Снимок экрана 2023-08-17 в 16 17 24

2. Moved `PromoBannerProps` interface to the related component file 
Снимок экрана 2023-08-17 в 16 12 19

3. Placed `barbershopData` object in the `src/Features/header/components/Promo/Promo.constants.ts`
Снимок экрана 2023-08-17 в 16 16 30

4. Wrote a tests for `src/components/Modal component` using `userEvent` Снимок экрана 2023-08-17 в 16 19 12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "typescript": "5.1.6"
   },
   "devDependencies": {
+    "@testing-library/dom": "^9.3.1",
```
Comment (Alex-2kZharkov) [inline package.json]: please check, I think you accidently installed a new library

File: src/components/Modal/Modal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  const clickCloseButton = jest.fn();
+
+  test('Should render correctly', () => {
+    render(<Modal showModal onClose={clickCloseButton} />);
+
+    expect(screen.getByRole('button')).toBeInTheDocument();
+  });
+
+  test('Handler is called 1 time', async () => {
+    const user = userEvent.setup();
```
Comment (Alex-2kZharkov) [inline src/components/Modal/Modal.test.tsx]: I don't think you need to do this extra step, check official docs https://testing-library.com/docs/ecosystem-user-event/#api

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ain-odin) [reply package.json]: Oops. My bad. Fixed it
Снимок экрана 2023-08-17 в 18 43 46

Comment (ain-odin) [reply package.json]: My bad. Fixed this moment
Снимок экрана 2023-08-17 в 18 43 46

File: src/components/Modal/Modal.test.tsx (hunk 2 above)
Comment (ain-odin) [reply src/components/Modal/Modal.test.tsx]: Got it
Снимок экрана 2023-08-17 в 18 43 59

File: package.json (hunk 1 above)
Comment (ain-odin) [reply package.json]: redone with `npm uninstall @testing-library/dom `

**Review by Alex-2kZharkov — APPROVED** [review]

**Pull request merged**